pymc_marketing/mmm/lift_test.py

```diff
@@ -0,0 +1,366 @@
+        dist: PyMC distribution to use for the likelihood.
+        model: PyMC model with date and channel coordinates.
+        name: Name of the likelihood.
+
+    Returns:
+        None
+
+    """
+    variable_mapping = {
+        "alpha": alpha_name,
+        "lam": lam_name,
+    }
+
+    add_lift_measurements_to_likelihood(
+        df_lift_test,
+        variable_mapping,
+        saturation_function=michaelis_menten,
+        model=model,
+        dist=dist,
+        name=name,
+    )
+
+
+def add_logistic_empirical_lift_measurements_to_likelihood(
+    df_lift_test: pd.DataFrame,
+    lam_name: str,
+    beta_name: str,
+    dist=pm.Gamma,
+    model: Optional[pm.Model] = None,
+    name: str = "lift_measurements",
+) -> None:
+    """Add empirical lift measurements to the likelihood of the model.
+
+    Args:
+        df_lift_test: DataFrame with lift test results.
+        lam_name: Name of the lambda parameter in the model.
+        beta_name: Name of the beta parameter in the model.
+        dist: PyMC distribution to use for the likelihood.
+        model: PyMC model with date and channel coordinates.
+        name: Name of the likelihood.
+
+    Returns:
+        None
+
+    """
+    variable_mapping = {
+        "lam": lam_name,
+        "beta": beta_name,
+    }
+
+    def saturation_function(x, beta, lam):
+        return beta * logistic_saturation(x, lam)
+
+    add_lift_measurements_to_likelihood(
+        df_lift_test,
+        variable_mapping,
+        saturation_function=saturation_function,
+        model=model,
+        dist=dist,
+        name=name,
+    )
+
+
+def scale_channel_lift_measurements(
+    df_lift_test: pd.DataFrame,
+    channel_col: str,
+    channel_columns: list[str],
+    transform: Callable[[np.ndarray], np.ndarray],
+) -> pd.DataFrame:
+    """Scale the lift measurements for a specific channel.
+
+    Args:
+        df_lift_test: DataFrame with lift test results.
+        channel_col: Name of the channel to scale.
+        channel_columns: List of channel names.
+        transform: Function to scale the lift measurements.
+
+    Returns:
+        DataFrame with the scaled lift measurements.
+
+    """
+    df_original = df_lift_test.loc[:, [channel_col, "x", "delta_x"]].set_index(
+        channel_col, append=True
+    )
+    df_to_rescale = (
+        df_original.stack()
+        .unstack(channel_col)
+        .reindex(channel_columns, axis=1)  # type: ignore
+        .fillna(0)
+    )
+
+    df_rescaled = pd.DataFrame(
+        transform(df_to_rescale.to_numpy()),
+        index=df_to_rescale.index,
+        columns=df_to_rescale.columns,
+    )
+
+    return (
+        df_rescaled.stack()
+        .unstack(1)
+        .loc[df_original.index, :]
+        .reset_index(channel_col)
+    )
+
+
+def scale_target_for_lift_measurements(
+    target: pd.Series,
+    transform: Callable[[np.ndarray], np.ndarray],
+) -> pd.Series:
+    """Scale the target for the lift measurements.
+
+    Args:
```

Comment:
I think we are following the numpy docstring format right? https://numpydoc.readthedocs.io/en/latest/format.html